Signl4 destination (#913)
rudderlabs/rudderstack-docs: docs/destinations/streaming-destinations/signl4.mdx

</div>

<div class="infoBlock">
Find the open source transformer code for this destination in the <a href="https://github.com/rudderlabs/rudder-transformer/tree/master/v0/destinations/signl4">GitHub repository</a>.
</div>

## Getting started

RudderStack supports sending events to SIGNL4 via the following <a href="https://rudderstack.com/docs/rudderstack-cloud/rudderstack-connection-modes/">connection modes</a>:

| **Connection Mode** | **Web** | **Mobile** | **Server** |
| :--- | :--- | :--- | :--- |
| **Device mode** | -  | - | - |
| **Cloud mode** | **Supported**  | **Supported** | **Supported** |

Once you have confirmed that the source platform supports sending events to SIGNL4, follow these steps:

1. From your [RudderStack dashboard](https://app.rudderstack.com/), add a source. Then, from the list of destinations, select **SIGNL4**.
2. Assign a name to the destination and click **Continue**.

## Connection settings

To successfully configure SIGNL4 as a destination, you will need to configure the following settings:

> - **API Key**: Enter the team secret of your SIGNL4 team. Refer to the <a href="#faq">FAQ</a> section to know about how to obtain the team secret.

Reviewer: ```suggestion
- **API Key**: Enter the team secret of your SIGNL4 team. Refer to the <a href="#faq">FAQ</a> section for more information on obtaining the team secret.
```

> - **X-S4-Service**: Enter a category for the alert, for example, IT. Refer to the <Link to="#signl-categories">Signl categories</Link> section to know more about these categories.

Reviewer: ```suggestion
- **X-S4-Service**: Enter a category for the alert, for example, `IT`. Refer to the <Link to="#signl-categories">Signl categories</Link> section to know more about these categories.
```

<img src="../../assets/event-stream-destinations/signl4-connection-settings-2.png" alt="SIGNL4 connection setting"/>

> - **X-S4-Location**: Enter the location in `latitude, longitude` format to display the correct location on the map in the mobile app.

Reviewer: ```suggestion
- **X-S4-Location**: Enter the location in the `latitude, longitude` format to display the correct location on the map in your mobile app.
```

<img src="../../assets/event-stream-destinations/signl4-connection-settings.png" alt="SIGNL4 connection setting"/>

> - **X-S4-Status**: Enter the alert status by choosing the **new**, **acknowledged**, or **resolved** values.

Reviewer: ```suggestion
- **X-S4-Status**: Enter the alert status by choosing from the **new**, **acknowledged**, or **resolved** values.
```

<img src="../../assets/event-stream-destinations/signl4-conn-settings-3.png" alt="SIGNL4 connection setting"/>

> - **X-S4-AlertingScenario**: Enter a value to control how SIGNL4 notifies the team:

Reviewer: ```suggestion
- **X-S4-AlertingScenario**: Use this setting to control how SIGNL4 notifies the team. RudderStack gives you the following options:
```

  - single_ack: Only one person needs to acknowledge the alert.
  - multi_ack: All on-duty persons need to acknowledge the alert. 
  - emergency: All persons, irrespective of their duty status, are notified and need to acknowledge the alert.
<img src="../../assets/event-stream-destinations/signl4-conn-settings-2.png" alt="SIGNL4 connection setting"/>
- **X-S4-ExternalID**: Use this field to pass the unique ID of a record if an event originates from that record in a third-party system. 
<img src="../../assets/event-stream-destinations/signl4-conn-settings-5.png" alt="SIGNL4 connection setting"/>

> As can be observed, RudderStack dashboard provides the following fields for each of the above-mentioned settings:

Reviewer: ```suggestion
The RudderStack dashboard contains the following fields for each of the above-mentioned settings:
```

> - **Default Value**: Enter the actual value for the field.

Reviewer: ```suggestion
- **Default Value**: Enter or select the required value for the field.
```

> - **RudderStack property name**: Enter a property name whose actual value will be taken from the payload and assigned to the field.

Reviewer: ```suggestion
- **RudderStack property name**: Enter the property name whose actual value will be taken from the event payload and assigned to the field.
```

> Although these fields are optional, a higher priority is given to the **RudderStack property name** if both are specified. For example:

Reviewer: ```suggestion
Although these fields are optional, RudderStack prioritizes **RudderStack property name** over **Default Value** if both are specified. For example, consider the following use-cases:
```

> - If you provide `new` in the **Default value** field for **X-S4-Status**, the output payload will contain `X-S4-Status: new`.

Reviewer: ```suggestion
- If you provide `new` in the **Default value** field for the **X-S4-Status** setting, the event payload sent to SIGNL4 will contain `X-S4-Status: new`.
```

> - If you provide `alert_status` in the **RudderStack property name** field, then the **X-S4-Status** value is taken from `properties.alert_status`. Suppose the value for `properties.alert_status` is set to `resolved`, the output payload will contain `X-S4-Status: resolved`.

Reviewer: ```suggestion
- If you set **RudderStack property name** for the **X-S4-Status** setting to `alert_status`, RudderStack looks for the event property `properties.alert_status`. Suppose the value for `properties.alert_status` is set to `resolved`, then the output payload sent to SIGNL4 will contain `X-S4-Status: resolved`.
```

> If you provide values in both fields, the final output payload will contain `X-S4-Status: resolved`, as the **RudderStack property name** is given the higher priority.

Reviewer: ```suggestion
- If you provide values in both the fields, the final output payload sent to SIGNL4 will contain `X-S4-Status: resolved`, as the **RudderStack property name** is given higher priority over **Default Field**.
```

> - **X-S4-Filtering**: Enable this setting and **Filter events/Signls** setting in the SIGNL4 dashboard to send an alert notification only if the input payload contains any keyword for a category. Refer to the <Link to="#signl-categories">Signl categories</Link> section to know more about keywords and category categories.

Reviewer: Can't find the **Filter events/Signls** setting in the below screenshot. Does toggling **Enable to add filter** enable this? 

If so, we can change the screenshot accordingly.

Author: nope, that setting is in the SIgnl4 dashboard, as mentioned. Also, there is a screenshot in **Signl categories** section and a reference is added here.

<img src="../../assets/event-stream-destinations/signl4-conn-settings-4.png" alt="SIGNL4 connection setting"/>

<div class="infoBlock">
Refer to the <a href="https://connect.signl4.com/webhook/docs/index.html">SIGNL4 documentation</a> to know more about these connection settings.
</div>

> - **Event to title mapping**: Enter the event name and alert title you want to be mapped corresponding to each other. If you do not provide any value, the alert title is set as the event name.

Reviewer: ```suggestion
- **Event to title mapping**: This setting lets you map the RudderStack event name to a specific alert title. If you do not provide any value, RudderStack sets the alert title as the event name.
```

## Signl categories

> In SIGNL4, you can use the categories to mark and tag Signls, for example, based on severity, type or relating IT systems, machines, sensors, and services. Refer to the [SIGNL4 documentation](https://account.signl4.com/manage/Category) to know more about different categories.

Reviewer: ```suggestion
In SIGNL4, you can use the categories to mark and tag Signls based on severity, type, or relating to IT systems, machines, sensors, and services. Refer to the [SIGNL4 documentation](https://account.signl4.com/manage/Category) to know more about the different categories.
```

> You can use the **X-S4-Service** setting in the RudderStack dashboard to specify the Signl category. If not provided, the keywords present in the payload decide the category of the alert. For example, an event payload containing keywords like `ambulance`, `doctor`, `hospital`, `nurse`, `surgery`, etc., is categorized as **Healthcare**.

Reviewer: ```suggestion
You can use the **X-S4-Service** setting in the RudderStack dashboard to specify the Signl category. If not provided, the keywords present in the payload decide the category of the alert. For example, an event payload containing keywords like `ambulance`, `doctor`, `hospital`, `nurse`, `surgery`, etc., is automatically categorized as **Healthcare**.
```

<img src="../../assets/event-stream-destinations/categories-signl4.png" alt="SIGNL4 connection setting"/>

## Track

> The <Link to="/event-spec/standard-events/track">`track`</Link> call can be used to push alerts for events with a particular alert title, alert message, or other additional details.

Reviewer: ```suggestion
You can use the <Link to="/event-spec/standard-events/track">`track`</Link> call to push alerts for events with a particular alert title, message, or other additional details.
```

A sample `track` call is shown below:

```javascript

> rudderanalytics.track(event, {

Reviewer: Should `event` have a specific name? e.g. `"Security Alert"` cc @ujjwal-ab

Reviewer 2: Ya, Let's add a specific event name. Let's add `Security Alert`.

> rudderanalytics.track(event, {

Reviewer: ```suggestion
rudderanalytics.track("Security Alert", {
```

  "status": "UP",

>   "message": "Agressive passenger",

Reviewer: ```suggestion
  "message": "Aggressive passenger",
```

  "Source": "Gate agent",
  "Type": "Security"
});
```

> You can pass any key-value pair in the `track` call; it is mapped as it is in the SIGNL4 destination.

Reviewer: ```suggestion
You can pass any key-value pair in the `track` call; RudderStack maps it as is in the SIGNL4 destination.
```

### Property mapping

The following table lists the mappings between RudderStack and SIGNL4 properties:

> | RudderStack property | SIGNL4 property | Presence | Deescription | 

Reviewer: ```suggestion
| RudderStack property | SIGNL4 property | Presence | Description | 
```

| :----------------- | :--------------- |:--------------- |:--------------- |
| `event`     | `title`      | Required | Title of the alert. |
| `property.message`   | `message`     | Optional  | Text message of the alert. |

## FAQ

### Where can I find the SIGNL4 API Key?

The team secret of your SIGNL4 team is the SIGNL4 API key in the RudderStack dashboard. To find the team secret, follow these steps:

1. Log into your [SIGNL4 dashboard](https://connect.signl4.com/).
2. Select **Teams** to find the **Secret**, as shown:

<img src="../../assets/event-stream-destinations/signl-dashboard-2.png" alt="SIGNL4 script url"/>

### How can I create a member in the SIGNL4 team?

To create a member in the SIGNL4 team, follow these steps:
1. Log into your [SIGNL4 dashboard](https://connect.signl4.com/).
2. Select **Users** > **Invite new user**, as shown:

> <img src="../../assets/event-stream-destinations/signl-dashboard-1.png" alt="SIGNL4 script url"/>

Reviewer: IMO this is a very basic FAQ, do you think we can eliminate the image here and just list the basic steps? Saves us some space 😄
